spec/features/user_feature_spec.rb

```diff
@@ -0,0 +1,95 @@
+# Flows
+# 1. Sign in 
+# Cannot view profile page without signing in
+# Cannot access any page other than signup page
+# Directs to Timeline page
+
+# 2. Sign up
+# Requires fields filled correctly
+# Fields filled correctly creates user and generates profile
+# Directs to profile page after signing up
+
+require 'rails_helper'
+
+describe User do
+	
+	let(:user){create(:user, :with_profile)}
+
+	before do
+		visit root_path
+	end
+
+	context 'sign in' do
+
+		scenario 'works with valid email and password' do
+			fill_in 'email', with:  user.email
+			fill_in 'password_digest', with:  user.password
+			click_button 'Log In'
+			expect(page).to have_content("Signed in successfully!")
+		end
+
+		scenario 'requires valid credentials' do
+		 click_button 'Log In'
+		 expect(current_url).to eq(root_url) 
+		 expect(page).to have_content('ERROR')
+		end
+
+		scenario 'redirects user profile on success' do
+			fill_in 'email', with:  user.email
+			fill_in 'password_digest', with: user.password
+			click_button 'Log In'
+			expect(current_url).to eq(user_profile_url(user))
+		end
+
+		scenario 'is required to access any page' do
+			visit user_profile_url(user)
+			expect(current_url).to eq(root_url)
+			expect(page).to have_css('.error-messages')
+		end
+
+	end
+
+	context 'sign up' do
+			
+		scenario 'requires unique email' do
```

Comment:
One thing you can do is pull this repeated code into a helper method.

Have the method accept a hash or **args

Then do something like

fill_in 'user_first_name', with: args[:first_name] || "James"

Then in the spec file you can just say
sign_up_user
or
sign_up_user(:password => "123423123")